File: pset6/cash.py
from cs50 import get_float

# Prompt user until valid (non-negative) input
while True:
    dollars = get_float("Change owed: ")
    if dollars >= 0:
        break

# Convert dollars to cents and round to avoid floating-point issues
cents = round(dollars * 100)

# Coin denominations in cents
coins = 0
for coin in [25, 10, 5, 1]:
    coins += cents // coin
    cents %= coin

# Output minimum number of coins
print(coins)

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;

    // Prompt user until a non-negative int is provided
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int coins = 0;

    // Greedy algorithm: Use largest coin first
    coins += cents / 25;
    cents %= 25;

    coins += cents / 10;
    cents %= 10;

    coins += cents / 5;
    cents %= 5;

    coins += cents; // Remaining 1s

    printf("%i\n", coins);
}
// This program calculates the minimum number of coins needed to make change for a given amount of
// cents. It uses a greedy algorithm, starting with the largest coin denomination (25 cents) and
// working down to the smallest (1 cent). The user is prompted to enter a non-negative integer
